zerver/actions/custom_profile_fields.py

```diff
@@ -209,10 +216,46 @@ def do_update_user_custom_profile_data_if_changed(
             },
         )
 
+        new_value = get_custom_profile_field_display_value(field_value)
+
+        changes.append(
+            UserProfileChangeDict(
+                field_name=field_value.field.name,
+                old_value=old_value,
+                new_value=new_value,
+            )
+        )
+
+    if changes and notify:
+        send_user_profile_update_notification(
+            user_profile=user_profile, acting_user=acting_user, changes=changes
+        )
+
+
+def get_custom_profile_field_display_value(field_value: CustomProfileFieldValue) -> str:
+    """Convert custom profile field value to human-readable string based on field type."""
+    if not field_value.value:
+        return ""
+    type = field_value.field.field_type
+
+    if type == CustomProfileField.SELECT:
+        field_data_dict = orjson.loads(field_value.field.field_data)
+        value_key = field_value.value
+        return field_data_dict[value_key]["text"]
+
+    elif type == CustomProfileField.USER:
+        user_ids = orjson.loads(field_value.value)
+        users_names = UserProfile.objects.filter(id__in=user_ids).values_list(
+            "full_name", flat=True
+        )
+        return (", ").join(users_names)
+    else:
+        return field_value.value
+
 
 @transaction.atomic(savepoint=False)
 def check_remove_custom_profile_field_value(
-    user_profile: UserProfile, field_id: int, acting_user: UserProfile
+    user_profile: UserProfile, field_id: int, acting_user: UserProfile, notify: bool = True
```

Comment:
I think this and other functions where we notify can also have default set to `False`.